src/rtsp_player_controller.cc

```diff
@@ -131,13 +131,25 @@ void RTSPPlayerController::InitPlayer(const std::string& url, const double& audi
 }
 
 void RTSPPlayerController::UpdateAudioConfig() {
-	//Hardcode settings to AAC
-	audio_config_.codec_type = Samsung::NaClPlayer::AUDIOCODEC_TYPE_AAC;
-	audio_config_.codec_profile = Samsung::NaClPlayer::AUDIOCODEC_PROFILE_AAC_LOW;
-	audio_config_.channel_layout = Samsung::NaClPlayer::CHANNEL_LAYOUT_MONO;
-	audio_config_.sample_format = Samsung::NaClPlayer::SAMPLEFORMAT_PLANARF32;
-	audio_config_.bits_per_channel = 6; // bitrate divided by sample rate
-	audio_config_.samples_per_second = 8000;
+	AVStream* s = format_context_->streams[audio_stream_idx_];
+	if(s->codecpar->codec_id == AV_CODEC_ID_AAC){
+		is_transcode=false;
+		audio_config_.codec_type = Samsung::NaClPlayer::AUDIOCODEC_TYPE_AAC;
+		audio_config_.codec_profile = ConvertAACAudioCodecProfile(s->codecpar->profile);// Method
+		audio_config_.channel_layout =  ConvertChannelLayout(s->codecpar->channel_layout, s->codecpar->channels); //Method
+		audio_config_.sample_format = ConvertSampleFormat(s->codec->sample_fmt);
+		audio_config_.bits_per_channel = av_get_bytes_per_sample(s->codec->sample_fmt) * 8 / s->codecpar->channels;
```

Comment:
`s->codec` is deprecated. can use `s->codecpar->format` here:

From docs:
- video: the pixel format, the value corresponds to enum AVPixelFormat.
- audio: the sample format, the value corresponds to enum AVSampleFormat.